File: 20.java
import java.util.Stack;

class Solution {
    public boolean isValid(String s) {
        Stack<Character> st = new Stack<>();
        for(char c: s.toCharArray()){
            if(c=='(' || c=='{' || c=='[')
            st.push(c);
            else{
                if(c==')' && !st.empty() && st.peek()=='(')
                st.pop();
                else if(c=='}' && !st.empty() && st.peek()=='{')
                st.pop();
                else if(c==']' && !st.empty() && st.peek()=='[')
                st.pop();
                else
                return false;
            }
        }

        return st.empty();       

    }
}

File: 20.cs
public class Solution {
    public bool IsValid(string s) {
        Stack<char> st = new Stack<char>();
        foreach (char c in s) {
            if (c == '(' || c == '{' || c == '[') {
                st.Push(c);
            } else {
                if (c == ')' && st.Count > 0 && st.Peek() == '(') {
                    st.Pop();
                } else if (c == '}' && st.Count > 0 && st.Peek() == '{') {
                    st.Pop();
                } else if (c == ']' && st.Count > 0 && st.Peek() == '[') {
                    st.Pop();
                } else {
                    return false;
                }
            }
        }
        return st.Count == 0;
    }
}